### ISSUE TITLE
Default Puppetlabs Ubuntu deb version is too old (precise)

### ISSUE THREAD

**Issue by themasterchef** [neillturner/kitchen-puppet#67]
## Problem

In puppet_apply.rb, we see that the default Puppet APT repo is very old (Ubuntu precise)

```
default_config :puppet_apt_repo, 'http://apt.puppetlabs.com/puppetlabs-release-precise.deb'
```

When using a Vagrant + Kitchen setup like this:

```
 platforms:
   - name: ubuntu-14.04
```

It fails to detect that '14.04' and sticks with the 'precise' URL. This breaks installation, because the `puppet` and `hiera-puppet` packages in this repo have broken dependencies:

```
The following packages have unmet dependencies:
        hiera-puppet : Depends: puppet but it is not going to be installed
       E: Unable to correct problems, you have held broken packages.
```

## Fix

Therefore it might be a good plan to update this default config to at least point at the trusty package.

## Workaround

In the meantime, this kitchen.yml setup seems to get past this problem stage:

```
provisioner:
  name: puppet_apply
  require_chef_for_busser: false
  puppet_apt_repo: http://apt.puppetlabs.com/puppetlabs-release-trusty.deb

platforms:
  - name: ubuntu-14.04
```

**Comment by grubernaut** [comment]
Just a quick update:

I have a patch to fix the repo issues automatically, so you no longer have to specify which apt-repo to use for the puppet install.

But the reason that the `hiera-puppet` install works from the `trusty` repo and not the `precise` repo, is that hiera was included in `puppet-common` from versions of puppet > 3.x.x. The `trusty` repo selects the correct package for you, while the `precise` repo does not.

Using the `trusty` repo:

```
Note, selecting 'puppet-common' instead of 'hiera-puppet'
puppet-common is already the newest version.
```

Using the `precise` repo:

```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 hiera-puppet : Depends: puppet but it is not going to be installed                                                                
E: Unable to correct problems, you have held broken packages.
```

If using a version of Puppet that is greater than 3.x.x, hiera is installed via the `puppet-common` package anyways.

This does mean, however, if you have a version of puppet that is older than 3.x.x and you attempt to install the `hiera-puppet` package, it will fail as documented above.

**Comment by neillturner** [comment]
this should be resolved now in version 1.0.32. Well for the puppet apply not the puppet agent.
